astropy/convolution/convolve.py

```diff
@@ -358,10 +358,10 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
             new_result = Kernel1D(array=result)
         elif isinstance(passed_array, Kernel2D):
             new_result = Kernel2D(array=result)
+        else:
+            raise TypeError("Only 1D and 2D Kernels are supported.")
         new_result._is_bool = False
         new_result._separable = passed_array._separable
-        if isinstance(passed_kernel, Kernel):
-            new_result._separable = new_result._separable and passed_kernel._separable
```

Comment:
This looks like a potentially important functional change - is it intentional?

Reply:
`new_result._separable` has already been set to ``passed_array._separable`` the line above. I don't see how this line could change it.

Reply:
This line is checking whether the `passed_kernel` is _also_ separable; it's different from `passed_array`.  That can be different, no?

Reply:
oh, sure. I can't read apparently (did double check it so many times, but always read passed_kernel as passed_array)...

Reply:
:+1: that's what code review is for